sockets.js

```diff
@@ -212,7 +215,43 @@ if (cluster.isMaster) {
 	}
 
 	let app = require('http').createServer();
-	let appssl = Config.ssl ? require('https').createServer(Config.ssl.options) : null;
+	let appssl = null;
+	if (Config.ssl) {
+		// HTTPS server options take buffers of private key and certificate
+		// files, but we save some bandwidth for IPC by sending their filenames
+		// and reading them here instead.
```

Comment:
...you're saving bandwidth for a once-per-restart event? In a way that actively restricts how easy it is to set up SSL?

Reply:
We don't even IPC them; we already read them every time in `config.js`. All you save is a read in Main and in other child processes that need sockets. Which can be done by just deferring them by supporting callbacks.

Reply:
In short, this is entirely unnecessary...

Reply:
This is all you need to minimize reads, if for whatever reason you cared:

`config.js`

```
exports.ssl = {
  get options: function () {
     // load keys here
  }
};
```

Reply:
It has more to do with the Go rewrite, where the config settings need to be passed to the Go process trhough IPC since it can't read .js files. There isn't much of a reason to change it like this otherwise

Reply:
In which case it still shouldn't affect the format of the `.ssl` object...

Reply:
Yeah, it's doable to still open the files from the main process and send it over with the rest of the config settings when first spawning the Go process. Just it'll be tricky to pass the PEM files over to it in a secure way. Doing it through environment variables like how the rest of the config settings are passed doesn't come across to me as a very secure way of dealing with it. I may need to refactor the part of it that handles the HTTP(S) servers so I could do something like send them over through its IPC connection

Reply:
But anyway, I can change this back. Is checking that the SSL config's valid fine to keep though?

Reply:
Good point... I guess passing a file location might be the best solution... but we need to do this in a backwards-compatible way.

Reply:
Like, support the old format for a while, so we have no downtime.

Reply:
Sounds good